packages/flutter/lib/src/widgets/icon.dart

```diff
@@ -247,6 +248,11 @@ class Icon extends StatelessWidget {
   /// [IconThemeData.applyTextScaling].
   final bool? applyTextScaling;
 
+  /// A blend mode to apply when a shape is drawn or a layer is composited.
```

Comment:
Maybe: The [BlendMode] to apply to the foreground of the icon.

Also instead of saying "Detailed in [Paint.blendMode].",  consider documenting the default, and does setting this to null do.

Reply:
```
The [BlendMode] to apply to the foreground of the icon.

When null, the default behavior is to use [color] as-is.
```

Thanks for the tips!

Reply:
It doesn't have to be nullable I think? so maybe make this non-nullable and say: "Defaults to [BlendMode.srcOver]"

Reply:
@LongCatIsLooong  sorry for the delay.

Are you referring to merging the color into the foreground?
I’ve already done this in commit [b1b222170](https://github.com/flutter/flutter/pull/150315/commits/b1b2221706107040a77549d765342588bb5e419c), but it broke several tests. I can try to fix them (might need some help), if this is what you were expecting?

Reply:
>   /// When null, the default behavior is to use [color] as-is.

This sounds a bit strange for describing a `BlendMode`. Maybe: "Defaults to [BlendMode.srcOver]". Also maybe the field should be non-nullable since the theme doesn't have a blendMode?

Reply:
I've change the description.

non-nullable means we need to merge the `color` and `blendMode` into `foreground`?

```dart
// from
Color? iconColor = color ?? iconTheme.color!;
Paint? foreground;
if (blendMode != null) {
  foreground = Paint()
    ..blendMode = blendMode!
    ..color = iconColor;
  // Cannot provide both a color and a foreground.
  iconColor = null;
}
// to
Color iconColor = color ?? iconTheme.color!;
Paint foreground = Paint()
    ..blendMode = blendMode!
    ..color = iconColor;
```